src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs

```diff
@@ -102,24 +100,39 @@ public JsonResult Create(TeamViewModel teamViewModel)
             {
                 try
                 {
-                    var roster = teamViewModel.Roster.Select(t => t.ToCreatePlayerDto()).ToList();
-
-                    var players = _playerService.CreateBulk(roster);
-
-                    var team = teamViewModel.ToCreateTeamDto();
-                    teamViewModel.Roster = players.Select(x =>
-                        new PlayerNameViewModel {
-                            Id = x.Id,
-                            FirstName = x.FirstName,
-                            LastName = x.LastName
-                        })
-                            .ToList();
-
-                    var createdTeam = _teamService.Create(team);
+                    if (teamViewModel.Captain.Id == 0)
+                    {
+                        var newAddedCaptain = _playerService.Create(teamViewModel.Captain.ToCreatePlayerDto());
+                        teamViewModel.Captain = new PlayerNameViewModel {
+                            Id = newAddedCaptain.Id,
+                            FirstName = newAddedCaptain.FirstName,
+                            LastName = newAddedCaptain.LastName
+                        };
+                        var team = teamViewModel.ToCreateTeamDto();
+                        var createdTeam = _teamService.Create(team);
+                        teamViewModel.Id = createdTeam.Id;
+                    }
+                    else
+                    {
+                        var team = teamViewModel.ToCreateTeamDto();
+                        var createdTeam = _teamService.Create(team);
+                        teamViewModel.Id = createdTeam.Id;
+                        createdTeam.SetCaptain(new PlayerId(teamViewModel.Captain.Id));
+                    }
+                    
+                    if (teamViewModel.AddedPlayers.Count > 0)
```

Comment:
What about removed players?

Reply:
it is create controller, probably there is no deleted players